src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContext.cs

```diff
@@ -75,7 +75,7 @@ public DependencyContext Merge(DependencyContext other)
                 );
         }
 
-        [RequiresAssemblyFiles(Message = "Calls GetDepsJsonPath")]
+        [RequiresAssemblyFiles(Message = "The use of DependencyContextLoader is not supported when publishing as single-file")]
```

Comment:
These should be user facing messages. While this is technically correct it doesn't help much to the user.

```C#
// IL3002: The use of DependencyContextLoader is not supported when publishing as single-file.
DependencyContext.Default.ToString();
```

User: What??? I'm not calling `DependencyContextLoader`, I'm calling `DependencyContext.Default`... this must be a bug - I'll ignore it.

The message should read something like:
"DependencyContext for an assembly from a application published as single-file is not supported. The method will return `null`. Make sure the calling code can handle this case."